### PR TITLE
[SPARK-28808][DOCS][SQL] Document SHOW FUNCTIONS in SQL Reference.

### PR DESCRIPTION
**Pull request by dilipbiswal**
### What changes were proposed in this pull request?
Document SHOW FUNCTIONS statement in SQL Reference Guide. 

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes. 

**Before:**

**After.**
image

Screen Shot 2019-09-04 at 11 41 44 AM

Screen Shot 2019-09-04 at 11 41 54 AM

### How was this patch tested?
Tested using jykyll build --serve

### CODE REVIEW

**Comment by dilipbiswal** [comment]
cc @gatorsmile

**Comment by sharangk** [comment]
@dilipbiswal , Should we add the below note?
"LIKE" qualifier is optional and has no relevance. It is allowed only for compatibility purpose.

File: docs/sql-ref-syntax-aux-show-functions.md [hunk 1]
```diff
@@ -19,4 +19,47 @@ license: |
...
+Three optional parameters `SYSTEM`, `USER`, `ALL` may be specified to list
+system functions, user defined function or both user and system defined
+funtions respectively. Given Spark supports hundreds of functions, this
+statement in conjuction with [describe function](sql-ref-syntax-aux-describe-function.html)
+may be used to quickly find the function and understand its usage.
+
+### Syntax
+{% highlight sql %}
+SHOW [USER|SYSTEM|ALL] FUNCTIONS ([LIKE] qualified_function_name | regex_pattern)
+qualified_function_name:= [db.]function_name
```
Comment (gatorsmile) [inline docs/sql-ref-syntax-aux-show-functions.md]: Is this still needed? How about only doing it in the section `### Parameters`?

**Comment by sharangk** [comment]
image

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master.

**Comment by dilipbiswal** [comment]
@gatorsmile Thank you very much.
@sharangk Thank you for your comments.

**Pull request closed without merging**